tests/tensorflow2/test_keras.py

```diff
@@ -395,14 +395,15 @@ def test_keras_fit(out_dir, tf_eager_mode, saveall):
     # can't save gradients in TF 2.x eager mode
     if saveall:  # save losses, metrics, weights, biases
         if tf_eager_mode:
-            assert len(trial.tensor_names()) == (7 if is_tf_2_2() else 8)
+            assert len(trial.tensor_names()) == (12 if is_tf_2_2() else 13)
         else:
             assert len(trial.tensor_names()) == 21
         assert len(trial.tensor_names(collection=CollectionKeys.BIASES)) == 2
         assert len(trial.tensor_names(collection=CollectionKeys.WEIGHTS)) == 2
     else:  # save the default losses and metrics
-        assert len(trial.tensor_names()) == (3 if is_tf_2_2() and tf_eager_mode else 4)
+        assert len(trial.tensor_names()) == (12 if is_tf_2_2() and tf_eager_mode else 13)
     assert len(trial.tensor_names(collection=CollectionKeys.LOSSES)) == 1
+    assert len(trial.tensor_names(collection=CollectionKeys.OPTIMIZER_VARIABLES)) == 5
```

Comment:
is this true in case of both, vanilla TF and custom TF?

Reply:
what's the code flow in case of vanilla TF? where is set_optimizer_variables called from?